### ISSUE TITLE
Life Module Issues

### ISSUE THREAD

**Issue by ghost** [chummer5a/chummer5a#1570]
### Environment
Chummer Version: 5.192.8
Environment: Windows 10
Runtime: .NET 4.5.1

### Expected behaviour
1. When adding any life modules that give you negative qualities, Your negative Karma stays the same (at least I found no rules saying you get 'positive' negative karma)

2. When removing a Life module that gives you a boosted attribute, the 'Base' should revert back to normal

3. When adding a life module that gives you Knowledge skills, Knowledge skills should be added AND additional points are given for life modules that don't dictate exactly what should be taken. A prompt should appear for the person to select the knowledge skill if applicable.

4. When adding a life module that gives you specializations, the specialization should be added to the skill. 

5. When adding the Formative Years, White Collar life module that gives a +1 Negotiation skill, it should display as rating one in skills. 

6. When adding Teen Years, Magical Education you should be prompted to select 2 skills groups, adept powers, or one/two of both if Mystic adept. 

### Actual behaviour
1. When selected an Life Module that gives you a negative quality (example: nationality UCAS and Teen Years, Magical Education), you are given 'positive' negative point equal to what the quality is. This allows you to take more than 25 points of negative qualities. This also reduces the Karma Remaining by the same amount of karma.

2. When removing a Life module that gives you a boosted attribute, the 'Base' does NOT revert back to normal. Example:  nationality UCAS. It gives +1 to Logic, but when you remove it the 'Base' still shows 2. 'Val (Aug)'  DOES properly change back.

3. When adding a life module that gives you Knowledge skills, no skills are actually added. Points are given equal to what should have been added. No prompt is given to select 'limited skills'; such as secondary language in Nationality Life Modules.

4. When adding a life module that gives you specializations, the specialization is NOT added to the skill.
Example: Formative Years, White Collar - Etiquette (Professional) specialization does not get added.

5. When adding the Formative Years, White Collar life module that gives a +1 Negotiation skill, you 'secretly' get the rating, but does NOT appear as it has any ratings UNLESS you manually add one rating. The one rating does not cost anything, and after adjusting it +1, you can't lower it as expected.
-This issue also occurs with State College, Business for all skills that you receive. Might happen on many others

6. When adding Teen Years, Magical Education you are not prompted to select any skills groups or adept powers. If you try to Manually select a skill group, you DO end up spending points.

### Steps to reproduce the behaviour
Create any new Character (I used Human and Elf for testing), and just follow and select the appropriate values listed in the issues.

### Link to affected character file
All files

**Comment by SpellBinder-883** [comment]
I too have noticed the karma count issue with qualities in the Life Modules system.  Currently I have an override file set where all karma values are set to zero as it seems that Chummer5 is adding the relevant qualities for free.

I've also seen the noted Attribute and skill issues, and have also noted a few other errors within the life modules XML.  As I recall, this system is still a bit of a mess to get cleaned up and running properly.

**Comment by SpellBinder-883** [comment]
Somewhat sure this has also already been mentioned, but just in case it hasn't, I noticed an incident where the Life Modules start tacking on spent karma on Active Skills.  Was making a human and had chosen the modules, in order, General UCAS, Part Of The Machine, Hacker Club, and then in the State University/Computer Science and noticed 10 karma suddenly added to Active Skills.  I had seen this happen in other Life Module made characters, and this time decided to investigate what was going on.

It seems that if you have skills in a broken skill group from one or more Life Modules, then add a module that adds a rank in the entire group itself, extra karma is automatically spent.  I tried changing the skill selection from Part Of The Machine to grant a rank in Electronics as an alternative (since the module gives individually 1 to all three skill in that group), and found that this just added to the cost in this scenario.  So far the only workaround I've found that'll work properly is to break up the skill groups in the life modules to reward those skills individually.
